tests/storage/blocksd_test.py

```diff
@@ -1643,6 +1643,66 @@ def test_create_illegal_volume(domain_factory, domain_version, fake_task,
     assert vol.getLegality() == sc.ILLEGAL_VOL
 
 
+@requires_root
+def test_reduce_volume_called(domain_factory, fake_task, fake_sanlock):
+    sd_uuid = str(uuid.uuid4())
+    dom = domain_factory.create_domain(sd_uuid=sd_uuid, version=5)
+
+    img_uuid = str(uuid.uuid4())
+    vol_id = str(uuid.uuid4())
+    vol_capacity = 10 * GiB
+    initial_size = 5 * GiB
+
+    dom.createVolume(
+        imgUUID=img_uuid,
+        capacity=vol_capacity,
+        volFormat=sc.COW_FORMAT,
+        preallocate=sc.SPARSE_VOL,
+        diskType=sc.DATA_DISKTYPE,
+        volUUID=vol_id,
+        desc="Over extended volume",
+        srcImgUUID=sc.BLANK_UUID,
+        srcVolUUID=sc.BLANK_UUID,
+        initial_size=initial_size)
+
+    # Prepare volume before reducing.
+    vol = dom.produceVolume(img_uuid, vol_id)
+    vol.prepare()
+    dom.reduceVolume(img_uuid, vol_id, allowActive=True)
+
+    # Check that (prepared) volume size has been reduced.
+    assert dom.getVolumeSize(
+        img_uuid, vol_id).apparentsize == vol.optimal_size()
```

Comment:
This is valid python, but harder to read. It will be nicer as:

    new_size = dom.getVolumeSize(img_uuid, vol_id).apparentsize
    assert new_size == vol.optimal_size()